src/autoProcess.ts

```diff
@@ -127,9 +127,7 @@ export function sveltePreprocess(
     }
 
     if (sourceMap && name in SOURCE_MAP_PROP_MAP) {
-      const [propName, value] = SOURCE_MAP_PROP_MAP[name];
-
-      opts[propName] = value;
+      setProp(opts, ...SOURCE_MAP_PROP_MAP[name]);
```

Comment:
Nice! However, I think something like this would be easier to understand:

```suggestion
      const [propPath, value] = SOURCE_MAP_PROP_MAP[name];
      setProp(opts, propPath, value);
```

Reply:
so verbose : (

i say its clear, when you see
```js
setProp(opts, ...SOURCE_MAP_PROP_MAP[name]);
```
and

```js
export function setProp(obj, keyList, val) {
```
and

```js
export const SOURCE_MAP_PROP_MAP: Record<string, [string[], any]> = {
  babel: [['sourceMaps'], true],
  typescript: [['compilerOptions', 'sourceMap'], true],
```
that 1+1+1==3